tstate/tstate_view.go

```diff
@@ -49,14 +52,30 @@ type TStateView struct {
 	writes      map[string]uint16
 }
 
-func (ts *TState) NewView(scope set.Set[string], storage map[string][]byte) *TStateView {
+func (ts *TState) NewView(scope set.Set[types.Key], storage map[string][]byte) *TStateView {
+	// TODO: set length to be the number of keys for each type
+	readScope := set.NewSet[string](len(scope))
+	writeScope := set.NewSet[string](len(scope))
+	rwScope := set.NewSet[string](len(scope))
```

Comment:
Not sure if it's better to have the scopes already separated and then passed into NewView or done within NewView (or if it makes a difference)